Azure Network August release

MikhailTryakhov opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

Merged ```Network-August-Release``` branch to ```master```

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

MikhailTryakhov committed:
Created folder for 2018-07-01 api version (#3509)

* Created folder for 2018-07-01 api version

* fixed incorrect encoding symbol

anton-evseev committed:
Fix lint diff and improve output on failure (#3528)

Lint diff was failing on PRs with a base branch other than master. We
needed to fetch the base branch ref from the origin before trying to
resolve it to its commit hash.

lmazuel committed:
Python conf for Network 2018-06 (#3535)

lmazuel committed:
Py conf for Network 2018-07-01 (#3539)

lmazuel committed:
Merge remote-tracking branch 'upstream/master' into Network-August-Release

anilingle-ms committed:
Introduce Publicipprefix to swagger (#3520)

* Add PublicIpPrefix

* Add PublicIpPrefix

* Add PublicIpPrefix

* Move to 2018-07-01 branch

* revertversionchange

* Fix Syntax error

* Fix IpTag description because evidently everything is global

* Fix $ref for PublicIpPrefix

* Fix Casing

* Add Exception for PublicIPPrefix from RequiredPropertiesMissingInResourceModel

* Fix Version#

* Fix Examples

anilingle-ms committed:
Publicipprefixaugust (#3573)

* Add PublicIpPrefix

* Add PublicIpPrefix

* Add PublicIpPrefix

* Move to 2018-07-01 branch

* revertversionchange

* Fix Syntax error

* Fix IpTag description because evidently everything is global

* Fix $ref for PublicIpPrefix

* Fix Casing

* Add Exception for PublicIPPrefix from RequiredPropertiesMissingInResourceModel

* Fix Version#

* Fix Examples

* Remove IdleTimeout parameter

* Fix comma

chandrasekarsrinivasan committed:
Service Endpoint Policies Swagger (#3558)

* Service Endpoint Policies Swagger

* Fix review comments

anton-evseev committed:
Updated Network Usage location parameter pattern (#3583)

dumahesh-msft committed:
Add enableTcpReset property to load balancer (#3545)

* Add enableTcpReset property to load balancer.

* Update loadbalancer examples with enableTcpReset.

syfarogh committed:
added global reach flag (#3542)

lmazuel committed:
Merge remote-tracking branch 'upstream/master' into Network-August-Release

junyezhu committed:
fix query packet capture status in latest version (#3601)

anilingle-ms committed:
Publicipprefixaugust (#3618)

* Add PublicIpPrefix

* Add PublicIpPrefix

* Add PublicIpPrefix

* Move to 2018-07-01 branch

* revertversionchange

* Fix Syntax error

* Fix IpTag description because evidently everything is global

* Fix $ref for PublicIpPrefix

* Fix Casing

* Add Exception for PublicIPPrefix from RequiredPropertiesMissingInResourceModel

* Fix Version#

* Fix Examples

* Remove IdleTimeout parameter

* Fix comma

* Add PublicIpPrefix for PublicIpAddressConfiguration

* Fix PublicIPAddress to refer to PublicIpPrefix as SubResource

* Fix Comma

* Remove compute update

anilingle-ms committed:
Publicipprefixaugust (#3631)

* Add PublicIpPrefix

* Add PublicIpPrefix

* Add PublicIpPrefix

* Move to 2018-07-01 branch

* revertversionchange

* Fix Syntax error

* Fix IpTag description because evidently everything is global

* Fix $ref for PublicIpPrefix

* Fix Casing

* Add Exception for PublicIPPrefix from RequiredPropertiesMissingInResourceModel

* Fix Version#

* Fix Examples

* Remove IdleTimeout parameter

* Fix comma

* Add PublicIpPrefix for PublicIpAddressConfiguration

* Fix PublicIPAddress to refer to PublicIpPrefix as SubResource

* Fix Comma

* Remove compute update

* Add PublicIpPrefix Subresource to VirtualMachineScaleSetPublicIPAddressConfigurationProperties

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-java

jianghaolu commented on specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/VirtualNetworkGatewayConnectionUpdateTags.json:
```diff
@@ -0,0 +1,47 @@
+          "virtualNetworkGateway1": {
+            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworkGateways/vpngw"
+          },
+          "localNetworkGateway2": {
```
According to https://github.com/Azure/azure-rest-api-specs/pull/3635/files#diff-795d13db3a317a9440687541d8d00549R2545, `properties` property is required.

MikhailTryakhov replied:
Guys all these changes were approved in previous PRs. this was just copy-pasted from previous api-version and had no problems as well. every CI job is passed. I see properties above. can't understand a problem

jianghaolu commented on specification/network/resource-manager/Microsoft.Network/stable/2018-07-01/examples/VirtualNetworkGatewayConnectionUpdateTags.json:
```diff
@@ -0,0 +1,47 @@
+          "provisioningState": "Succeeded",
+          "resourceGuid": "00000000-0000-0000-0000-000000000000",
+          "virtualNetworkGateway1": {
+            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworkGateways/vpngw"
```
According to https://github.com/Azure/azure-rest-api-specs/pull/3635/files#diff-795d13db3a317a9440687541d8d00549R2545, `properties` property is required.

pashukl committed:
Changes to Support MSEEv2 Datapath for ExpressRoute. (#3647)

Description of Changes:
    - Added a flag expressRouteGatewayBypass to the Connection
      Properties.
    - The user can set the flag to True to use MSEEv2 datapath

Related work items: #2628003

irrogozh committed:
Added new API to networkwatcher.json and examples (#3650)

* Added new API to networkwatcher.json and examples

* Added NetworkWatcherNetworkConfigurationDiagnostic example

MikhailTryakhov commented:
@jianghaolu can you please merge it?

chandrasekarsrinivasan committed:
Fix name from ServiceEndpointDefinition to ServiceEndpointDefinitions while doing a put, get and delete (#3655)

khannarhea committed:
Add/Edit Outbound Rule on Load Balancer and Examples (#3592)

* Add/Edit Outbound Rule on Load Balancer and Examples

* Adding properties enabletcpreset, idletimeoutinminutes and protocol

* Adding a reference to Public IP Prefix

* Dummy Commit

irrogozh committed:
Removed QueryConnectionMOnitors API (#3661)

* Added new API to networkwatcher.json and examples

* Added NetworkWatcherNetworkConfigurationDiagnostic example

* removed QueryConnectionMonitors Api
